Benchmarks

Nmargolis opened the pull request:
### Context
This PR adds some basic benchmark infrastructure.
Issue #3.

### Summary of changes
- [x] Adds Criterion benchmarks for Coalesce
- [x] Adds test util for loading grids from a line-delimited json file

### Next steps (future)
- [ ] Download a larger, more realistic dataset from s3 or other location
- [ ] Potentially add a [script](https://medium.com/@yamafaktory/rust-benchmarking-with-criterion-on-travis-ci-%EF%B8%8F-8b54d321e05) to run these benchmarks against master on CI

apendleton commented on benches/coalesce.rs:
```diff
@@ -0,0 +1,238 @@
+    let mut to_bench = Vec::new();
+
+    // Load data for coalesce single from json into a store
+    let filepath = Path::new("benches/data/coalesce-bench-single-3848571113.json");
```
What is this relative to? Does it matter where you run `cargo bench` from?

Nmargolis replied:
It's relative to the output of `std::env::current_dir()` which I think is the absolute path to the `carmen-core` dir. This seems to work regardless of whether the bench command is run in `carmen-core/` or `carmen-core/rust-src/` or `carmen-core/benches`, which suggests it's not dependent on where you run the bench command from. I added a comment to the load_grids_from_json function that the path should be relative to `carmen-core/`

apendleton commented on rust-src/src/gridstore/store.rs:
```diff
@@ -246,9 +246,6 @@ impl GridStore {
                             .unwrap();
-                    // TODO could this be a reference? The compiler was saying:
-                    // "cannot move out of captured variable in an `FnMut` closure"
-                    // "help: consider borrowing here: `&match_opts`rustc(E0507)""
```
Why remove these comments? (Mostly asking because this PR doesn't otherwise change this file, and it's going to cause a merge conflict with #37)

Nmargolis replied:
Just because they seem like they shouldn't be in there. Agreed that it's not relevant to this PR, so reverting. At some point we can do a separate sweep for TODOs that probably don't belong in the code.

Nmargolis commented:
Added a comment about what the file path should be relative to, and reverted the removal of TODOs in an unrelated file.

The concern about using the same case over and over is valid and we should definitely have a more realistic set of queries to run. For now, this is essentially a port of the similarly-simplistic [carmen-cache bench](https://github.com/mapbox/carmen-cache/blob/master/bench/coalesce.bench.test.js)
